File: 0371-sum-of-two-integers/0371-sum-of-two-integers.go
/*
func getSum(a int, b int) int {
    if b == 0 {
        return a
    }
    return getSum(a^b, (a&b)<<1)
}
*/

func getSum(a int, b int) int {
    for b != 0 {
        a, b = (a^b), (a&b)<<1
    }
    return a
}

File: 0371-sum-of-two-integers/0371-sum-of-two-integers.py
class Solution:
    def getSum(self, a: int, b: int) -> int:
        carry = 0
        mask = 0xffffffff
        while b & mask != 0:
            a, b = a ^ b, (a & b) << 1
            
        if mask < b:
            return a & mask
        return a